common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -827,13 +833,16 @@ void resetChunkTracker() {
     void updateChunkInfo(long chunkOffset, int mapIndex) throws IOException {
       long idxStartPos = -1;
       try {
-        // update the chunk tracker to meta file before index file
-        writeChunkTracker(mapIndex);
         idxStartPos = indexFile.getFilePointer();
         logger.trace("{} shuffleId {} reduceId {} updated index current {} updated {}",
           appShuffleId.appId, appShuffleId.shuffleId, reduceId, this.lastChunkOffset,
           chunkOffset);
-        indexFile.writeLong(chunkOffset);
+        indexFile.write(Longs.toByteArray(chunkOffset));
```

Comment:
I wish `Longs` took a parameter - we can skip allocation/clear of the array (cheap as it might be).

Reply:
I vaguely remember there is a utility for this somewhere either in Spark code or Hadoop code. I tried searching for it in Spark but couldn't find it.

Reply:
There are multiple cases of this in spark; but nothing which is reusable. That is ok, the cost is not very high, and this is infrequent enough in comparison to be an issue :)